### PR TITLE
Fix java client

### PR DESCRIPTION
**Pull request by darionyaphet**
#816 [graph client for java exists problems](https://github.com/vesoft-inc/nebula/issues/816)

When an error occurs in query execution, will throw an `NGQLException`

### CODE REVIEW

File: src/client/java/src/main/java/com/vesoft/nebula/graph/client/GraphClientImpl.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        }
+    }
+
+    @Override
+    public int executeUpdate(String stmt) {
+        return execute(stmt);
+    }
+
+    @Override
+    public ResultSet executeQuery(String stmt) throws NGQLException {
```
Comment (zlcook) [inline src/client/java/src/main/java/com/vesoft/nebula/graph/client/GraphClientImpl.java]: NGQLException doesn't seem to be enough. In addition to `ResultSet`, `executeQuery` should contain these status values that `execute` method will return.

File: src/client/java/src/main/java/com/vesoft/nebula/graph/client/GraphClientImpl.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(GraphClientImpl.class.getName());
+
+    private int retry;
+    private final int timeout;
+    private long sessionId_;
+    private TTransport transport = null;
+    private GraphService.Client client;
+    private List<InetSocketAddress> nodes = Lists.newLinkedList();
+
+    public GraphClientImpl(Map<String, Integer> addresses, int timeout, int retry) {
```
Comment (zlcook) [inline src/client/java/src/main/java/com/vesoft/nebula/graph/client/GraphClientImpl.java]: If multiple instances with different ports are running on the same ip (Ps, ip1:3307, ip1:3308), the `addresses` map can only save one.

Comment (darionyaphet) [reply src/client/java/src/main/java/com/vesoft/nebula/graph/client/GraphClientImpl.java]: That's strange.

**Comment by zlcook** [comment]
LGTM

File: src/client/java/src/main/java/com/vesoft/nebula/graph/client/GraphClientImpl.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
```
Comment (dangleptr) [inline src/client/java/src/main/java/com/vesoft/nebula/graph/client/GraphClientImpl.java]: Why use map here?

**Comment by darionyaphet** [comment]
@dangleptr Currently you could get result code or exception use to check and reconnect.

**Comment by jude-zhu** [comment]
Jenkins go

File: src/client/java/pom.xml [hunk 4]
```diff
@@ -71,8 +71,6 @@
             <groupId>com.facebook</groupId>
             <artifactId>thrift</artifactId>
             <version>1.0.0-SNAPSHORT</version>
```
Comment (zlcook) [inline src/client/java/pom.xml]: type: SNAPSHOT

File: src/client/java/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+
+### Java Client
+
+***
+
+The Java Client is dependent on `com.facebook:thrift:1.0.0-SNAPSHORT`.
```
Comment (zlcook) [inline src/client/java/README.md]: typo: SNAPSHOT

File: src/client/java/build.sh [hunk 6]
```diff
@@ -51,10 +53,9 @@ setup_graph_source()
 #-----------------------------------------------
 compile_java_client()
 {
-	mvn clean package -DJAVA_FBTHRIFT_JAR=$java_fbthrift_jar -f ${pom_file_path}
```
Comment (zlcook) [inline src/client/java/build.sh]: If mvn repo doesn't exist `fbthrift` dependency, `Java-client` can't be compiled successfully. So should add `mvn install::install-file ...` command to install `fbthrift` into local mvn repo before this command.

Comment (darionyaphet) [reply src/client/java/build.sh]: I  have add this in read.me `mvn install ` is ok.

**Comment by darionyaphet** [comment]
After talk with @laura-ding offline, we will add `mvn install` when compile third party library.

**Comment by laura-ding** [comment]
please add instruction in `how_to_build.md` when build java client failed because of missing thrift, the user can install `/opt/nebula/third-party/fbthrift/thrift-1.0-SNAPSHOT.jar` to resolve it.

**Comment by zlcook** [comment]
LGTM

**Comment by darionyaphet** [comment]
jenkins go

**Comment by darionyaphet** [comment]
jenkins go

**Comment by darionyaphet** [comment]
jenkins go

**Comment by darionyaphet** [comment]
jenkins go

**Pull request merged**